xquant/keras/keras_report_utils.py

```diff
@@ -0,0 +1,380 @@
+            for key in metrics:
+                metrics[key].append(results[key])
+
+        # Averaging metrics across the dataset
+        aggregated_metrics = {key: sum(value) / len(value) for key, value in metrics.items()}
+
+        return aggregated_metrics
+
+    def get_metric_on_intermediate(self,
+                                   float_model: keras.Model,
+                                   quantized_model: keras.Model,
+                                   dataset: Callable,
+                                   custom_metrics_intermediate: Dict[str, Callable] = None,
+                                   is_validation: bool = False) -> Dict[str, Dict[str, float]]:
+        """
+        Compute metrics on intermediate layers of the model.
+
+        Args:
+            float_model (keras.Model): The floating-point Keras model.
+            quantized_model (keras.Model): The quantized Keras model.
+            dataset (Callable): Dataset used for inference.
+            custom_metrics_intermediate (Dict[str, Callable], optional): Custom metrics for intermediate layers.
+            Defaults to None.
+            is_validation (bool, optional): Flag indicating if this is a validation dataset. Defaults to False.
+
+        Returns:
+            Dict[str, Dict[str, float]]: A dictionary of computed metrics for intermediate layers.
+        """
+
+        float_model = self.create_float_folded_model(float_model=float_model,
+                                                     representative_dataset=None)
+
+        dataset = partial(self.wrapped_dataset,
+                          dataset=dataset,
+                          is_validation=is_validation)
+
+        def get_activations(model: Model,
+                            layer_names: List[str],
+                            data: Any) -> Dict[str, np.ndarray]:
+            """
+            Extract activations from specified layers of the model for the given input data.
+
+            Args:
+                model (Model): The Keras model from which to extract activations.
+                layer_names (List[str]): List of layer names for which activations are to be extracted.
+                data (Any): Input data for which activations are to be computed.
+
+            Returns:
+                Dict[str, np.ndarray]: A dictionary mapping layer names to their corresponding activations.
+            """
+            # Create a new model that outputs the activations of the specified layers
+            intermediate_layer_model = Model(inputs=model.input,
+                                             outputs=[model.get_layer(name).output for name in layer_names])
+
+            predictions = intermediate_layer_model.predict(data)
+
+            # Map the activations to their corresponding layer names
+            return {layer_name: predictions[i] for i, layer_name in enumerate(layer_names)}
+
+        metrics_to_compute = list(self.get_default_metrics().keys())
```

Comment:
this part is duplicated, maybe extract it to a util function "collect_metrics_to_compute" or something like that